utils/frame/sub-storage/src/lib.rs

```diff
@@ -0,0 +1,422 @@
+//! # Sub-Storage.
+//!
+//! A thing wrapper around substrate's RPC calls that work with storage. This module is an
+//! equivalent ot the polkadojs-api `api.query` and `api.const`, but written in Rust.
+//!
+//! This crate is heavily dependent on the `jsonspsee` crate and uses it internally to connect to
+//! nodes.
+//!
+//!
+//! The base functions of this crate make no assumption about the runtime. Some runtime-dependent
+//! functions are provided under the `helpers` module.
+//!
+//! ## Unsafe RPC calls.
+//!
+//! The most useful features provided by this crate are often marked as unsafe by the substrate
+//! nodes. Namely, [`get_pairs`] and [`enumerate_map`] can only be used against nodes that such
+//! external RPCs.
+//!
+//! THIS IS A TEST.
+
+use codec::Decode;
+use frame_support::StorageHasher;
+use sp_core::hashing::twox_128;
+use std::fmt::Debug;
+
+use jsonrpsee_http_client::{HttpClient, HttpConfig};
+use jsonrpsee_ws_client::{WsClient, WsConfig};
+use jsonrpsee_types::jsonrpc::{Params, to_value as to_json_value};
+
+/// Helper's module.
+#[cfg(feature = "helpers")]
+pub mod helpers;
+
+/// re-export some stuff from sp-core.
+pub use sp_core::storage::{StorageData, StorageKey};
+/// The hash type used by this crate.
+pub type Hash = sp_core::hash::H256;
+// TODO: write a basic abstraction above the two?
+pub type Client = WsClient;
```

Comment:
it looks like you can remove this and have `impl jsonrpsee::core::client::ClientT` in the helpers instead of the concrete type.

jsonrpsee has a client trait which both clients implements but it's not object-safe i.e, can't be used a trait object.